sql/src/main/java/io/crate/analyze/AlterTableAnalyzer.java

```diff
@@ -56,6 +57,23 @@ public AlterTableAnalyzedStatement analyze(AlterTable node, Row parameters, Sess
         return new AlterTableAnalyzedStatement(docTableInfo, partitionName, tableParameter, table.excludePartitions());
     }
 
+    AlterTableRenameAnalyzedStatement analyzeRename(AlterTableRename node, String defaultSchema,
+                                                    SessionContext sessionContext) {
+        TableIdent tableIdent = TableIdent.of(node.table(), defaultSchema);
+        DocTableInfo tableInfo = schemas.getTableInfo(tableIdent, sessionContext.user());
+        if (!node.table().partitionProperties().isEmpty()) {
+            throw new UnsupportedOperationException("Renaming a single partition is not supported");
+        }
+
+        List<String> newIdentParts = node.newName().getParts();
+        if (newIdentParts.size() > 1) {
+            throw new IllegalArgumentException("Target table name must not include a schema");
```

Comment:
Is there a (technical) constraint that we don't allow this? If so - maybe add a short code comment to mention it